### ISSUE TITLE
Incorrect result for else/elif

### ISSUE THREAD

**Issue by pkolbus** [Melevir/cognitive_complexity#11]
Thanks for the great package and flake8 plugin.

I've noticed that else (and nested elif) seem to be counted incorrectly. Based on my reading of the whitepaper (section B3, the discussion of "hybrid" increments, `overriddenSymbolFrom()` and other examples in section C, and the whitepaper's change log), `else` and `elif` are not subject to the nesting increment.

Sample test cases:
```
# passes
def test_nested_if_condition_complexity():
    assert get_code_snippet_compexity("""
    def f(a, b):
        if a == b:  # +1
            if (a):  # +2 (nesting=1)
                return 1
        return 0
    """) == 3

# fails (actual=1, expected=2)
def test_simple_else_condition_complexity():
    assert get_code_snippet_compexity("""
    def f(a):
        if (a):  # +1
            return 1
        else:  # +1
            return 3
    """) == 2

# fails (actual=3, expected=4)
def test_nested_else_condition_complexity():
    assert get_code_snippet_compexity("""
    def f(a, b):
        if a == b:  # +1
            if (a):  # +2 (nesting=1)
                return 1
            else:  # +1
                return 3
        return 0
    """) == 4

# fails  (actual=6, expected=5)
def test_nested_elif_condition_complexity():
    assert get_code_snippet_compexity("""
    def f(a, b):
        if a == b:  # +1
            if (a):  # +2 (nesting=1)
                return 1
            elif (b):  # +1
                return 2
            else:  # +1
                return 3
        return 0
    """) == 5
```
I'm not terribly familiar with Python AST parsing. But if you like, I can try working on a PR if you want (and agree it's an actual bug).

**Comment by Melevir** [comment]
Thank you for your kind words and for the report.
It really seems like a bug.
Problem is I don't have time to fix it right now, so it would be great if you submit the pull request.

**Comment by pkolbus** [comment]
Apologies for the delay; I've been pulled to other projects the last few weeks (both personal and professional). Finally got a chance to work on cognitive complexity again this weekend, and found the cause.

First, `ast.iter_child_nodes()` just yields the members of `orelse` without indicating the `else` itself. Second, an `elif` was incrementing the nesting level because it looked no different than a nested `if`. This means that a simple `if..else` is undercounted by 1, and a simple `if..elif..else` is correct but for the wrong reasons (+2 to the `elif` when it should be +1 to the `elif` and +1 to the `else`).

Since `for` and `while` have optional `else` clauses, I've attempted to account for those as well.

I'll create the PR momentarily.

@Melevir, I also noticed two other cases where complexity seems to be overcounted. Since the fix involves changes in the same areas, I can add these to the PR I'm working on, or write separate issues and PRs if you prefer.

- I think `ast.BoolOp` should not receive a nesting increment (`test_real_function()` should have a cognitive complexity of `9`)
- I think `break` and `continue` should not receive any increment (unlike C and Java there is no `goto` and no `break`/`continue` _to a label_, and no increment is given in the `addVersion` example in the Cognitive Complexity white paper).
